### PR TITLE
Add sanity tests for SSP alerts

### PR DESCRIPTION
**Pull request by assafad**
This PR adds new test cases for SSP alert definitions. We expect to see `summary`, `runbook_url`, `severity`, `kubernetes_operator_part_of` and `kubernetes_operator_component` for each alert. Those fields must not be empty and some must have specific values.

Signed-off-by: assafad <aadmi@redhat.com>

```release-note
None
```

### CODE REVIEW

File: tests/tests_common_test.go [hunk 1]
```diff
@@ -394,3 +396,35 @@ func addDomainResourcesToVMI(vmi *kubevirtv1.VirtualMachineInstance, cpuCores ui
 	}
 	return vmi
 }
+
```
Comment (akrejcir) [inline tests/tests_common_test.go:399]: All these functions are only used in one place, can you inline them?

File: tests/tests_common_test.go [hunk 2]
```diff
@@ -394,3 +396,35 @@ func addDomainResourcesToVMI(vmi *kubevirtv1.VirtualMachineInstance, cpuCores ui
...
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have runbook_url annotation", rule.Alert))
+	resp, err := http.Head(url)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("%s runbook is not available", rule.Alert))
+	ExpectWithOffset(1, resp.StatusCode).Should(Equal(http.StatusOK), fmt.Sprintf("%s runbook is not available", rule.Alert))
+}
+
+func checkForSummary(rule promv1.Rule) {
+	summary, ok := rule.Annotations["summary"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
+	ExpectWithOffset(1, summary).ToNot(BeEmpty(), fmt.Sprintf("%s has an empty summary", rule.Alert))
```
Comment (akrejcir) [inline tests/tests_common_test.go:411]: Can you replace this with `HaveKeyWithValue()` matcher? For example:
```go
Expect(rule.Annotations).To(HaveKeyWithValue("summary", Not(BeEmpty())), fmt.Sprintf("%s does not have summary annotation", rule.Alert))
```

File: tests/tests_common_test.go [hunk 3]
```diff
@@ -394,3 +396,35 @@ func addDomainResourcesToVMI(vmi *kubevirtv1.VirtualMachineInstance, cpuCores ui
...
+}
+
+func checkForSeverityLabel(rule promv1.Rule) {
+	severity, ok := rule.Labels["severity"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have severity label", rule.Alert))
+	ExpectWithOffset(1, severity).To(BeElementOf("info", "warning", "critical"), fmt.Sprintf("%s severity label is not valid", rule.Alert))
```
Comment (akrejcir) [inline tests/tests_common_test.go:417]: The same as above, can you use `HaveKeyWithValue()`? For example:
```go
Expect(rule.Labels).To(HaveKeyWithValue("severity",
   BeElementOf("info", "warning", "critical"),
))
```

File: tests/tests_common_test.go [hunk 4]
```diff
@@ -394,3 +396,35 @@ func addDomainResourcesToVMI(vmi *kubevirtv1.VirtualMachineInstance, cpuCores ui
...
+}
+
+func checkForPartOfLabel(rule promv1.Rule) {
+	kubernetesOperatorPartOf, ok := rule.Labels["kubernetes_operator_part_of"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have kubernetes_operator_part_of label", rule.Alert))
+	ExpectWithOffset(1, kubernetesOperatorPartOf).To(Equal("kubevirt"), fmt.Sprintf("%s kubernetes_operator_part_of label is not valid", rule.Alert))
```
Comment (akrejcir) [inline tests/tests_common_test.go:423]: You could use:
```go
Expect(rule.Labels).To(HaveKeyWithValue("kubernetes_operator_part_of", "kubevirt"))
```

File: tests/tests_common_test.go [hunk 5]
```diff
@@ -394,3 +396,35 @@ func addDomainResourcesToVMI(vmi *kubevirtv1.VirtualMachineInstance, cpuCores ui
...
+}
+
+func checkForComponentLabel(rule promv1.Rule) {
+	kubernetesOperatorComponent, ok := rule.Labels["kubernetes_operator_component"]
+	ExpectWithOffset(1, ok).To(BeTrue(), fmt.Sprintf("%s does not have kubernetes_operator_component label", rule.Alert))
+	ExpectWithOffset(1, kubernetesOperatorComponent).To(Equal("ssp-operator"), fmt.Sprintf("%s kubernetes_operator_component label is not valid", rule.Alert))
```
Comment (akrejcir) [inline tests/tests_common_test.go:429]: Similarly:
```go
Expect(rule.Labels).To(HaveKeyWithValue("kubernetes_operator_component", "ssp-operator"))
```

**Review by akrejcir — CHANGES_REQUESTED** [review]

**Comment by akrejcir** [comment]
Can you add more text to commit message?

File: tests/tests_common_test.go (hunk 1 above)
Comment (assafad) [reply tests/tests_common_test.go]: Done

**Pull request merged**